src/test/rgw/test_redis_lock.cc

```diff
@@ -0,0 +1,175 @@
+#include <gtest/gtest.h>
+#include <rgw_redis_lock.h>
+
+#include <boost/asio/deferred.hpp>
+#include <boost/asio/detached.hpp>
+#include <boost/asio/io_context.hpp>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "common/async/blocked_completion.h"
+
+using boost::redis::config;
+using boost::redis::connection;
+
+class RGWRedisLockTest : public ::testing::Test {
+ protected:
+  boost::asio::io_context io;
+  connection* conn;
+  config* cfg;
+
+  void SetUp() {
+    // Creating the default config
+    cfg = new config;
```

Comment:
better to use a `unique_ptr` and avoid the `delete`.
there are always excpetions, but in modern (11 and up)  cpp, a `delete` is considered a "red flag" and requires  a reason